src/scripts/main.js

```diff
@@ -1,3 +1,49 @@
 'use strict';
 
-// write your code here
+const promise1 = new Promise((resolve, reject) => {
+  const logo = document.querySelector('.logo');
+
+  if (logo) {
+    logo.addEventListener('click', () => {
+      resolve();
+    });
+  } else {
+    reject(new Error('Logo not found on the page'));
+  }
+});
```

Comment:
It's a good practice to move DOM manipulation and event listener assignment to a separate function. This would improve the readability of your code and make it more maintainable.